lib/Sema/DerivedConformanceCodable.cpp

```diff
@@ -647,31 +647,24 @@ deriveBodyEncodable_encode(AbstractFunctionDecl *encodeDecl, void *) {
   if (classDecl && superclassIsEncodable(classDecl)) {
     // Need to generate `try super.encode(to: container.superEncoder())`
 
-    // superEncoder()
-    auto *method = UnresolvedDeclRefExpr::createImplicit(C, C.Id_superEncoder);
-
-    // container.superEncoder()
-    auto *superEncoderRef = new (C) DotSyntaxCallExpr(containerExpr,
-                                                      SourceLoc(), method);
-
-    // encode(to:) expr
-    auto *encodeDeclRef = new (C) DeclRefExpr(ConcreteDeclRef(encodeDecl),
-                                              DeclNameLoc(), /*Implicit=*/true);
+    // container.superEncoder
+    auto *superEncoderRef = new (C) UnresolvedDotExpr(
+        containerExpr, SourceLoc(), DeclNameRef(C.Id_superEncoder),
+        DeclNameLoc(), /*Implicit=*/true);
 
     // super
     auto *superRef = new (C) SuperRefExpr(encodeDecl->getImplicitSelfDecl(),
                                           SourceLoc(), /*Implicit=*/true);
 
     // super.encode(to:)
-    auto *encodeCall = new (C) DotSyntaxCallExpr(superRef, SourceLoc(),
-                                                 encodeDeclRef);
+    auto *encodeRef = new (C) MemberRefExpr(superRef, SourceLoc(), encodeDecl,
+                                            DeclNameLoc(), /*Implicit=*/true);
```

Comment:
I feel like this original `DotSyntaxCallExpr(superRef, SourceLoc(),  encodeDeclRef)` is not right.
Shouldn't be `DotSyntaxCallExpr(encodeDeclRef, SourceLoc(), superRef)`?

Reply:
> Shouldn't it be `DotSyntaxCallExpr(encodeDeclRef, SourceLoc(), superRef)`?

Ouch. Yes, it looks like it should be.